assets/src/hooks/useChannel.ts

```diff
@@ -81,38 +81,86 @@ export const useCheckedChannel = <T, U>({
   loadingState: U
   closeAfterFirstRead?: boolean
 }): U => {
+  const [results, _push_updates] = useCheckedTwoWayChannel({
+    socket,
+    topic,
+    event,
+    dataStruct,
+    parser,
+    loadingState,
+    closeAfterFirstRead,
+  })
+  return results
+}
+
+export const useCheckedTwoWayChannel = <T, U, V>({
+  socket,
+  topic,
+  event,
+  dataStruct,
+  parser,
+  loadingState,
+  closeAfterFirstRead,
+}: {
+  socket: Socket | undefined
+  topic: string | null
+  event: string
+  dataStruct: Struct<T, any>
+  parser: (data: T) => U
+  loadingState: U
+  closeAfterFirstRead?: boolean
+}): [U, (event: string, payload: V) => void] => {
   const [state, setState] = useState<U>(loadingState)
+  const [joinedChannel, setJoinedChannel] = useState<Channel | undefined>()
+
+  const onOk = useCallback(
+    ({ data: data }: { data: unknown }) => {
+      try {
+        assert(data, dataStruct)
+        setState(parser(data))
+      } catch (error) {
+        if (error instanceof StructError) {
+          Sentry.captureException(error)
+        }
+      }
+    },
+    [dataStruct, parser, setState]
+  )
+
+  const pushUpdate = useCallback(
+    (event: string, payload: any): void => {
+      if (joinedChannel) {
+        joinedChannel
+          .push(event, payload)
+          .receive("ok", (data: { data: unknown }) => {
+            onOk(data)
+          })
+          .receive("error", ({ reason }) => {
+            // eslint-disable-next-line no-console
```

Comment:
**nitpick:** looks like this was left over from a version that logged to the console and this is no longer needed